x/ecocredit/simulation/genesis.go

```diff
@@ -165,92 +191,185 @@ func RandomizedGenState(simState *module.SimulationState) {
 	simState.GenState[ecocredit.ModuleName] = bz
 }
 
+func createClass(ctx context.Context, sStore api.StateStore, class *api.Class) (uint64, error) {
+	cKey, err := sStore.ClassTable().InsertReturningID(ctx, class)
+	if err != nil {
+		return 0, err
+	}
+
+	seq, err := sStore.ClassSequenceTable().Get(ctx, class.CreditTypeAbbrev)
+	if err != nil {
+		if ormerrors.IsNotFound(err) {
+			if err := sStore.ClassSequenceTable().Save(ctx, &api.ClassSequence{
+				CreditTypeAbbrev: class.CreditTypeAbbrev,
+				NextSequence:     2,
+			}); err != nil {
+				return 0, err
+			}
+			return cKey, nil
+		}
+
+		return 0, err
+	} else {
+		if err := sStore.ClassSequenceTable().Update(ctx, &api.ClassSequence{
+			CreditTypeAbbrev: class.CreditTypeAbbrev,
+			NextSequence:     seq.NextSequence + 1,
+		}); err != nil {
+			return 0, err
+		}
+	}
+
+	return cKey, nil
+}
+
+func createProject(ctx context.Context, sStore api.StateStore, project *api.Project) (uint64, error) {
+	pKey, err := sStore.ProjectTable().InsertReturningID(ctx, project)
+	if err != nil {
+		return 0, err
+	}
+
+	seq, err := sStore.ProjectSequenceTable().Get(ctx, project.ClassKey)
+	if err != nil {
+		if ormerrors.IsNotFound(err) {
+			if err := sStore.ProjectSequenceTable().Save(ctx, &api.ProjectSequence{
+				ClassKey:     project.ClassKey,
+				NextSequence: 2,
+			}); err != nil {
+				return 0, err
+			}
+			return pKey, nil
+		}
+
+		return 0, err
+	}
+
+	if err := sStore.ProjectSequenceTable().Update(ctx, &api.ProjectSequence{
+		ClassKey:     project.ClassKey,
+		NextSequence: seq.NextSequence + 1,
+	}); err != nil {
+		return 0, err
+	}
+
+	return pKey, nil
+}
+
+func getBatchSequence(ctx context.Context, sStore api.StateStore, projectKey uint64) (uint64, error) {
+	seq, err := sStore.BatchSequenceTable().Get(ctx, projectKey)
+	if err != nil {
+		if ormerrors.IsNotFound(err) {
+			if err := sStore.BatchSequenceTable().Save(ctx, &api.BatchSequence{
+				ProjectKey:   projectKey,
+				NextSequence: 2,
+			}); err != nil {
+				return 0, err
+			}
+			return 1, nil
+		}
+		return 0, err
+	}
+
+	if err := sStore.BatchSequenceTable().Save(ctx, &api.BatchSequence{
```

Comment:
We want to update an existing table, so we should use `Update` here as you have done above. `Save` will create a new table if one does not already exist with the `ProjectKey` provided.

```suggestion
	if err := sStore.BatchSequenceTable().Update(ctx, &api.BatchSequence{
```

Reply:
Save doesn't create a new table, it just runs `Insert` if there is no entry in the `BatchSequenceTable` yet, otherwise it will run `Update`. Using the more explicit function does make sense though.

Reply:
I meant to say insert a new entry, no "create a new table".

Reply:
Correct. I meant to say insert a new entry, not "create a new table".